### PR TITLE
Migrate from Browserify to Webpack2

### PR DESCRIPTION
**Pull request by jwineman**
To test:
`rm -rf node_modules/`
`npm install`

Commands:
- build
- build:production
- lint
- format
- test

build has sourcemaps, production is minified without sourcemaps.

### CODE REVIEW

**Comment by jwineman** [comment]
I'm going to modify this so webpack is a local dependency we can version.

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  watch: isDev,
+  module: {
+    loaders: [
+      {
+        test: /\.js$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        query: {
+          presets: ['es2015']
```
Comment (thellimist) [inline webpack.config.js]: https://github.com/cloudflare/cloudflare-plugin-frontend/blob/master/.babelrc

I don't think we need `presets ` we already have .babelrc. I'm not sure if the whole loader field is necessary.

Alternatively, we can delete .babelrc, and `babelrc: false` to webpack config.

File: package.json [hunk 2]
```diff
@@ -4,10 +4,11 @@
   "description": "A React/Redux frontend for Cloudflare plugins",
   "main": "gulpfile.js",
   "scripts": {
-    "test": "mocha src/js/test/*/*.js --compilers js:babel-register",
-    "lint": "gulp lint",
-    "start": "http-server -a localhost -p 8000 -c-1",
-    "postinstall": "bash git-hooks/setup.sh"
```
Comment (thellimist) [inline package.json]: why did we remove this?

Comment (jwineman) [inline package.json]: Because Travis CI runs the linter and tests now.  Don't need them running twice.

File: webpack.config.js (hunk 1 above)
Comment (jwineman) [inline webpack.config.js]: 1614b2d0e4b63a3c70548e271cb9d0c44a53e08f

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  plugins.push(new webpack.optimize.UglifyJsPlugin({
+      compress: isDev,
+      mangle: false
+    })
+  );
+}
+
+module.exports = {
+  entry: "./src/js/index.js",
+  devtool: isDev ? 'cheap-module-source-map' : false,
```
Comment (jwineman) [inline webpack.config.js]: If this devtool doesn't work for you can we can find one that does.

File: package.json (hunk 2 above)
Comment (jwineman) [inline package.json]: So I thought we were already doing this, opened a PR for Travis CI in https://github.com/cloudflare/cloudflare-plugin-frontend/pull/89.   `npm run build` runs the linter now too.

**Comment by thellimist** [comment]
README file should be edited.

Tested. It looks LGTM

**Pull request merged**